src/query/service/src/pipelines/processors/transforms/aggregator/transform_aggregate_partial.rs

```diff
@@ -240,22 +241,18 @@ impl<Method: HashMethodBounds> TransformPartialAggregate<Method> {
             .unwrap_or_default();
 
         let block = block.convert_to_full();
-
-        let group_columns = self
-            .params
-            .group_columns
-            .iter()
-            .map(|&index| block.get_by_offset(index))
-            .map(|c| (c.value.as_column().unwrap().clone(), c.data_type.clone()))
-            .collect::<Vec<_>>();
+        let group_columns = InputColumns::new_block_proxy(&self.params.group_columns, &block);
+        let data_types = group_columns.as_block().unwrap().data_types();
```

Comment:
This does not avoid the overhead of generating a `Vec`, so no performance improvement.